### PR TITLE
Remove `fail-fast` matrix strategy from GitHub Actions

### PR DESCRIPTION
**Pull request by clayburn**
## Description

Currently, every GitHub Actions workflow contains the `fail-fast` matrix strategy. However, none of these workflows are actually utilizing a matrix. Further, the on demand workflow has an improper indentation of this strategy block that causes a syntactical error.

This change remove the no-op strategy altogether to fix the broken on demand workflow and remove the IDE warnings regarding the missing matrix declaration.

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Scheduler
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [x] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md#coding-standards) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: .github/workflows/0-on-demand.yaml [hunk 1]
```diff
@@ -65,8 +65,6 @@ jobs:
   openwhisk:
     runs-on: ubuntu-22.04
     continue-on-error: false
-      strategy:
```
Comment (style95) [inline .github/workflows/0-on-demand.yaml:68]: I was trying to fix this in this PR: https://github.com/apache/openwhisk/pull/5420/files

**Comment by clayburn** [comment]
From the linked documentation on `fail-fast`: 

> jobs.<job_id>.strategy.fail-fast applies to the entire matrix. If jobs.<job_id>.strategy.fail-fast is set to true, GitHub will cancel all in-progress and queued jobs in the matrix if any job in the matrix fails. This property defaults to true.

This `strategy` block is specifically used to create a [matrix strategy](https://docs.github.com/ko/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategy) in order to run a job multiple times with different parameters. For example, you may do something like this to run a job against multiple operating systems:

```
strategy:
  fail-fast: true
  matrix:
    os: [ubuntu-latest, windows-latest]
```

In this case, if say `ubuntu-latest` were to fail, _then_ windows would be cancelled. But since you are not using a matrix strategy like this, the `fail-fast` would be a no-op.

I also get this warning in my IDE, although I do not know where to locate the schema it is using for this validation:

Screenshot 2023-06-21 at 8 09 55 AM

**Review by dgrove-oss — APPROVED** [review]
LGTM

**Comment by style95** [comment]
@clayburn Got it.
Thank you for the explanation.
I confirmed a build with the current master immediately stops on an error.
https://github.com/apache/openwhisk/pull/5423

**Pull request merged**